packages/core/src/api.ts

```diff
@@ -193,19 +193,17 @@ export const deploy = async (
     })
   }
 
-  const postDeployAction = async (appliedChanges: ReadonlyArray<Change>): Promise<void> => {
-    // create updated elements before updating the state, to prevent any unwanted behavior
-    const updatedElements = awu(appliedChanges)
+  const postDeployAction = async (appliedChanges: ReadonlyArray<Change>): Promise<void> => log.time(async () => {
+    // update changedElements before updating the state because we rely on the state data for the calculation
+    const updatedElements = await awu(appliedChanges)
       .filter(change => (isAdditionOrModificationChange(change) || isFieldChange(change)))
-      .map(getUpdatedElement)
+      .map(getUpdatedElement).toArray()
```

Comment:
be consistent with the newlines
```suggestion
      .map(getUpdatedElement)
      .toArray()
```